app/components/transUnit/TransUnitService.js

```diff
@@ -170,7 +198,26 @@
        updateSaveButton);
 
     function modifyTranslationText(phrase, newText) {
-      phrase.newTranslation = newText;
+      if (phrase.plural) {
+        var transUnitCtrl = controllerList[phrase.id];
+        phrase.newTranslations[transUnitCtrl.focusedTranslationIndex] = newText;
+      } else {
+        phrase.newTranslation = newText;
+      }
+      EventService.emitEvent(EventService.EVENT.TRANSLATION_TEXT_MODIFIED,
+        phrase);
+      EventService.emitEvent(EventService.EVENT.FOCUS_TRANSLATION,
+        phrase);
+    }
+
+    function modifyTranslationsText(phrase, newTexts) {
+      if(!phrase.plural) {
+        return; //only accept plural
```

Comment:
This will silently fail if someone writes bad code that calls the wrong function. We should at least log an error so that the coder calling the wrong function can see why it is not working.

A more complete fix would be to make the function work regardless whether the phrase is a plural or not, which would be part of the suggestion in my earlier comment.